stem/model.py

```diff
@@ -129,6 +132,95 @@ def add_soil_layer_by_coordinates(self, coordinates: Sequence[Sequence[float]],
 
         self.body_model_parts.append(body_model_part)
 
+    def add_load_by_coordinates(self, coordinates: Sequence[Sequence[float]], load_parameters: LoadParametersABC, name: str):
+        """
+        Adds a load to the model by giving a sequence of 3D coordinates. For a 2D model, the third coordinate is
+        ignored.
+
+        Args:
+            - coordinates (Sequence[Sequence[float]]): The coordinates of the load.
+            - parameters (:class:`stem.load.LoadParametersABC`): The parameters of the load.
+            - name (str): The name of the load part.
+
+        """
+
+        self.__validate_coordinates(coordinates)
+
+        if isinstance(load_parameters, PointLoad):
+            gmsh_input = {name: {"coordinates": coordinates, "ndim": 0}}
+        elif isinstance(load_parameters, get_args(Union[LineLoad, MovingLoad])):
```

Comment:
get_args is used because otherwise, mypy was complaining on the Union[LineLoad, MovingLoad]. The proposed solution work but it starts to be cumbersome if you have more than 2 types.

Reply:
allright